### ISSUE TITLE
Issue #298 - User Simulation Not Working

### ISSUE THREAD

**Issue by tomtinkerer** [LESEM-PEWI/pewi3.0#298]

**Comment by tomtinkerer** [comment]
Alex this one is for you

**Comment by Xemnaes** [comment]
Currently debugging and fixing.

To repeat this error:
- Go into sandbox and randomize the map
- After randomizing, begin recording and make changes to the map
- Stop recording and save the file
- Repeat simulation in utilities
**ERROR:** Steps are not repeated in simulator.

Also under this issue, the pause menu does not currently work while the simulation is underway. That too is being fixed.

**Comment by Xemnaes** [comment]
**UPDATE:** This bug only appears when a previous randomization occurs prior to recording.

**Comment by Xemnaes** [comment]
![image](https://user-images.githubusercontent.com/26779242/29185544-28bb2ef8-7dcf-11e7-8783-bcfc9e319b33.png)

The culprit has been found...looks like the _randomizing_ variable (the one which is true when the map is being randomized) was commented out, so that once the map is randomizing, the variable stays true. Doing so prevented any land uses to be painted on the map (since the program is told the map is still being randomized).

**Comment by Xemnaes** [comment]
Second issue is now fixed.

![image](https://user-images.githubusercontent.com/26779242/29186218-5f70e4e0-7dd1-11e7-8cb2-31333509892b.png)

Since the inclusion of navigational controls in the user sim, only program-generated inputs (including button presses) were allowed while the sim was running, which prevent the user from being able to pause. I have added a case of the escape key, which now allows the user to pause the sim.
